src/pykeen/models/base.py

```diff
@@ -502,160 +381,21 @@ def predict_tails(
-            The indices of (relation, tail) pairs.
-        :param slice_size: >0
-            The divisor for the scoring function when using slicing.
-
-        :return: shape: (batch_size, num_entities), dtype: float
-            For each r-t pair, the scores for all possible heads.
-        """
-        # Enforce evaluation mode
-        self.eval()
-
-        if self.triples_factory.create_inverse_triples:
-            scores = self.score_h_inverse(rt_batch=rt_batch, slice_size=slice_size)
-        elif slice_size is None:
-            scores = self.score_h(rt_batch)
-        else:
-            scores = self.score_h(rt_batch, slice_size=slice_size)
-        if self.predict_with_sigmoid:
-            scores = torch.sigmoid(scores)
-        return scores
-
-    def _score_all_triples(
-        self,
-        batch_size: int = 1,
-        return_tensors: bool = False,
-        *,
-        add_novelties: bool = True,
-        remove_known: bool = False,
-        testing: Optional[torch.LongTensor] = None,
-    ) -> Union[Tuple[torch.LongTensor, torch.FloatTensor], pd.DataFrame]:
-        """Compute and store scores for all triples.
-
-        :return: Parallel arrays of triples and scores
-        """
-        # initialize buffer on cpu
-        scores = torch.empty(self.num_relations, self.num_entities, self.num_entities, dtype=torch.float32)
-        assert self.num_entities ** 2 * self.num_relations < (2 ** 63 - 1)
-
-        for r, e in itt.product(
-            range(self.num_relations),
-            range(0, self.num_entities, batch_size),
-        ):
-            # calculate batch scores
-            hs = torch.arange(e, min(e + batch_size, self.num_entities), device=self.device)
-            hr_batch = torch.stack([
-                hs,
-                hs.new_empty(1).fill_(value=r).repeat(hs.shape[0]),
-            ], dim=-1)
-            scores[r, e:e + batch_size, :] = self.predict_scores_all_tails(hr_batch=hr_batch).to(scores.device)
-
-        # Explicitly create triples
-        triples = torch.stack([
-            torch.arange(self.num_relations).view(-1, 1, 1).repeat(1, self.num_entities, self.num_entities),
-            torch.arange(self.num_entities).view(1, -1, 1).repeat(self.num_relations, 1, self.num_entities),
-            torch.arange(self.num_entities).view(1, 1, -1).repeat(self.num_relations, self.num_entities, 1),
-        ], dim=-1).view(-1, 3)[:, [1, 0, 2]]
-
-        # Sort final result
-        scores, ind = torch.sort(scores.flatten(), descending=True)
-        triples = triples[ind]
-
-        if return_tensors:
-            return triples, scores
-
-        rv = self.make_labeled_df(triples, score=scores)
-        return _postprocess_prediction_all_df(
-            df=rv,
-            add_novelties=add_novelties,
-            remove_known=remove_known,
-            training=self.triples_factory.mapped_triples,
-            testing=testing,
-        )
-
-    def score_all_triples(
+    def get_prediction_df(
```

Comment:
Maybe use something like `get_every_prediction_df` or `get_all_prediction_df`? To me the current name is vague.

Reply:
I think I like `get_all_prediction_df` better. Will change this after resolving with master so I only have to run tests once